packages/legend-query-builder/src/stores/QueryLoaderState.ts

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DEFAULT_TYPEAHEAD_SEARCH_MINIMUM_SEARCH_LENGTH,
+  DEFAULT_TYPEAHEAD_SEARCH_LIMIT,
+  type GenericLegendApplicationStore,
+} from '@finos/legend-application';
+import {
+  type LightQuery,
+  QuerySearchSpecification,
+  type AbstractPureGraphManager,
+  type QueryProjectCoordinates,
+  V1_PureGraphManager,
+  type Query,
+  type QueryInfo,
+} from '@finos/legend-graph';
+import {
+  ActionState,
+  type GeneratorFn,
+  assertErrorThrown,
+  guaranteeType,
+  deleteEntry,
+  guaranteeNonNullable,
+} from '@finos/legend-shared';
+import { makeObservable, observable, action, flow, flowResult } from 'mobx';
+import type { QueryBuilderState } from './QueryBuilderState.js';
+import type {
+  LoadQueryFilterOption,
+  QueryBuilder_LegendApplicationPlugin_Extension,
+} from './QueryBuilder_LegendApplicationPlugin_Extension.js';
+
+export class QueryLoaderState {
+  readonly applicationStore: GenericLegendApplicationStore;
+  readonly loadQueriesState = ActionState.create();
+  queries: LightQuery[] = [];
+  isQueryLoaderOpen = false;
+  searchText = '';
+  showCurrentUserQueriesOnly = false;
+  showPreviewViewer = false;
+  queryPreviewContent?: QueryInfo;
+  extraFilters = new Map<string, boolean>();
+  extraFilterOptions: LoadQueryFilterOption[] = [];
+  searchSpecificationProjectCoordinates?: QueryProjectCoordinates[];
+  currentUserId: string | undefined;
+  queryBuilderState?: QueryBuilderState | undefined;
+  intialQueries: string[];
+  onDeleteQuery?:
+    | ((
+        intialQueries: string[],
+        idx: number,
+        applicationStore: GenericLegendApplicationStore,
+      ) => void)
+    | undefined;
+  onLoadQuery?:
+    | ((
+        intialQueries: string[],
+        val: LightQuery | Query,
+        applicationStore: GenericLegendApplicationStore,
+      ) => void)
+    | undefined;
+
+  constructor(
+    applicationStore: GenericLegendApplicationStore,
+    intialQueries?: string[],
```

Comment:
--> `fetchInitialQueries`

Can all of these method not pass around `intialQueries: string[]`? That's information is not needed to be passed around, technically, we only need 1 method from `QueryEditorStore` and the `initialQueryIds` will be known there, what' the point of passing this around all the methods?

Reply:
These methods are defined in the state of `QueryEditorStore` hence they need this information. The reason i haven't defined these methods in the editorStore is we need this logic for `SetupStores` for landing page as well then i would end up duplicating the logic. So i have just decided to have them as independent helper functions

Reply:
I think this is one of the use case where the `QueryBaseStore` might be useful? But it's ok, let's just not pass in a list of IDs like this, instead, let's do `fetchInitialQueries`